### PR TITLE
Shuffle NFTs 

### LINKED ISSUE #460: Election resolution with NFT shuffling

### PR DESCRIPTION
**Pull request by leomassazza**
Fix #460

### CODE REVIEW

File: packages/core-modules/contracts/modules/CoreElectionModule.sol [hunk 1]
```diff
@@ -313,15 +318,37 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
...
+
+        if (!isEpochFinished()) {
+            revert EpochNotFinished();
+        }
+
+        if (!isElectionEvaluated()) {
+            revert ElectionNotEvaluated();
+        }
+
+        for (uint i = 0; i < currentMembersSize; i++) {
```
Comment (leomassazza) [inline packages/core-modules/contracts/modules/CoreElectionModule.sol:336]: Back to burn all and mint again to simplify the logic (and reduce the chance of errors) to transfer the right token and prevent re-minting unburned tokens

File: packages/core-modules/contracts/modules/CoreElectionModule.sol [hunk 2]
```diff
@@ -313,15 +318,37 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
...
+        uint currentMembersSize = store.members.length;
+        uint nextEpochMembersSize = nextEpochMembers.length;
+
+        if (!isEpochFinished()) {
+            revert EpochNotFinished();
+        }
+
+        if (!isElectionEvaluated()) {
+            revert ElectionNotEvaluated();
+        }
```
Comment (ajsantander) [inline packages/core-modules/contracts/modules/CoreElectionModule.sol:334]: Can we perform these two checks before line 322 to save reversion gas?

**Review by ajsantander — APPROVED** [review]
Super clean!

File: packages/core-modules/contracts/modules/CoreElectionModule.sol (hunk 2 above)
Comment (leomassazza) [reply packages/core-modules/contracts/modules/CoreElectionModule.sol:334]: Done, and please check this decision.

Added the code after the checking to an internal function that later can be implemented in mixings for the different strategies.

**Review by i-stam — APPROVED** [review]

**Pull request merged**